crates/ursa-rpc-service/src/server.rs

```diff
@@ -32,15 +32,20 @@ where
         }
     }
 
-    pub async fn start(&self, config: &ServerConfig, metrics: Option<Router>) -> Result<()> {
+    pub async fn start(
+        &self,
+        config: &ServerConfig,
+        index_provider: Router,
+        metrics: Option<Router>,
+    ) -> Result<()> {
         info!(
             "Server (rpc, http{}) starting up",
             if metrics.is_some() { " + metrics" } else { "" }
         );
 
         let http_address = SocketAddr::from(([0, 0, 0, 0], config.port));
 
-        let service = MultiplexService::new(self.http_app(metrics), self.rpc_app());
+        let service = MultiplexService::new(self.http_app(metrics), self.rpc_app(index_provider));
```

Comment:
index_provider server should be an http endpoint, and not rpc

Reply:
perfect, will make this change :+1:

Reply:
https://github.com/fleek-network/ursa/pull/233/commits/43e8e00c4a8b9f53b9bee70a86542b41b68d5ad3